servicex/query_core.py

```diff
@@ -228,11 +228,16 @@ def transform_complete(task: Task):
                 logger.info(f"Transforms finished with code {self.current_status.status}")
 
         sx_request = self.transform_request
+        sx_request_hash = sx_request.compute_hash()
+
+        # Invalidate the cache if the hash already present but if the user ignores cache
+        if self.ignore_cache and self.cache.contains_hash(sx_request_hash):
```

Comment:
Any reason not to just do the check always - `if self.cache.contains_hash(sx_request_hash)` instead? That is - don't protect it by `ignore_cache`?

Reply:
If the user submits the same request again, then the cache will already have the record, so it will recompute rather than use the record from the cache. We only want to invalidate if the record is present but the user still wants to recompute (ignore cache).

Reply:
Also, I have not the subsequent logic, so I found it best to include this at a top without breaking anything below